### PR TITLE
Add torch_xla basic tests to onPR/onPush, return initialize_device_connectors, require tests to manually init xla computation client

### PR DESCRIPTION
**Pull request by jameszianxuTT**
### Ticket
#1556

### What's changed
- return `initialize_device_connectors` autouse fixture
  - remove  `torch_xla._XLAC._init_computation_client()` call from TorchDeviceConnector ctor, to avoid possible hangs when running pytorch and jax tests in the same process.
  - This means that tests will need to initialize the torch xla runtime manually, and most already do so by acquiring a device. Maybe this can be considered a reopening of #1383, since we don't have a good solution to that yet
- Add push marks to torchxla basic and torchxla custom ops tests
  - Due to --forked invocation ( #1554 ) add additional parallel splits to these test jobs so they don't take 1.5 hours to run
- xfail 26 torchxla basic ops tests failing due to PCC errors with unknown passing point #1555    
- In TorchDeviceRunner only move workload.compiled_executable .to device if .to is defined on the compiled_executable (not defined for customOps OpTester tests)

### Checklist
- [ ] New/Existing tests provide coverage for changes

### CODE REVIEW

File: tests/infra/connectors/torch_device_connector.py [hunk 1]
```diff
@@ -19,8 +19,6 @@ class TorchDeviceConnector(DeviceConnector):
     def __init__(self) -> None:
         super().__init__()
         xr.runtime.set_device_type("TT")
-        # Initialize TT PJRT computation client.
-        torch_xla._XLAC._init_computation_client()
```
Comment (jameszianxuTT) [inline tests/infra/connectors/torch_device_connector.py:23]: > This means that tests will need to initialize the torch xla runtime manually, and most already do so. Maybe this can be considered a reopening of https://github.com/tenstorrent/tt-xla/issues/1383, since we don't have a good solution to that yet

Per this comment - We probably need a better way to guarantee computation client initialization within test infra but putting that as a side effect of TorchDeviceConnector does not work since not all torch_xla tests use it. This caused a breaking change for torch_xla basic tests which was not noticed since they are not in CI

Solving this with an automatic test fixture that optionally initializes the computation client for torchxla tests doesn't seem like the best option. We'd still need to enforce that test groups running within the same process don't accidentally include both torchxla and jax tests. 

I think something like a custom fixture added over all torchxla tests that initializes the runtime if not initialized already might work, plus guards to check if jax is initialized in the same context [similar to how pytorch/xla does it](https://github.com/pytorch/xla/blob/1ab678799f2a9a448d202df0262a9cca9ea97bb1/torch_xla/_internal/jax_workarounds.py#L8-L12) might be a good option. We'd also need the reverse guard for jax (which I'm more unfamiliar with) to disallow initialization if torch_xla is detected to be initialized.

Comment (jameszianxuTT) [reply tests/infra/connectors/torch_device_connector.py:23]: I think the failure sequence you observed was specifically for running .backwards() as shown in reproducer of #1383. 

In that sequence:
1. set xr.runtime.set_device_type("tt")
2. run some operations including .backwards()
3. acquire device via xm.xla_device and implicitly cause runtime initialization
4. move weights/inputs to device
5. execute test
6. run .backwards() for ondevice graph and see autograd crash

I think the failure is related to autograd in (2) doing some weird stuff with XLA side effects, maybe causing gradients to get stuck on CPU or something so (6) fails. Based on findings in #1383 this seems to be a real bug, and unexpected that performing torch operations before acquiring a device has breaking side effects. i.e. doing steps in order (1-2-3 should be the same as doing them in order 1-3-2). It seems like it's only for training that this ordering difference matters.
 
The fix in #1339 was to enforce step order (1-3 ... ) before everything else by putting it in the TorchDeviceConnector initializer, but this seems only necessary for the particular training test and breaks other tests that don't construct a TorchDeviceConnector at start of test.

So, because this solution seems training-specific and breaks many other tests, I think it should be reverted as I have done in my PR. This means that for training graphs for now, test writer must remember to initialize device before running torch operations on CPU due to the bug described in https://github.com/pytorch/xla/issues/4174. 

Maybe in the future we can implement some guarded framework auto-initialization feature to each test, but for sake of getting torch_xla basic tests into CI sooner rather than later I suggest a simple reversion for now.

**Review by jmcgrathTT — APPROVED** [review]

**Review by nsumrakTT — APPROVED** [review]

**Comment by jameszianxuTT** [comment]
Merging this now.
cc @jazpurTT as it adds conv2d tests which may complicate your uplift.
